src/themes/teams/components/Divider/dividerStyles.ts

```diff
@@ -2,37 +2,28 @@ import * as _ from 'lodash'
 
 import { childrenExist } from '../../../../lib'
 import { pxToRem } from '../../utils'
-import { ComponentSlotStylesInput, ICSSInJSStyle, ICSSPseudoElementStyle } from '../../../types'
+import { ComponentSlotStylesInput, ICSSInJSStyle } from '../../../types'
 import { DividerPropsWithDefaults } from '../../../../components/Divider/Divider'
+import { DividerVariables } from './dividerVariables'
 
-const dividerBorderStyle = (size, color): ICSSInJSStyle => ({
-  height: `${size + 1}px`,
-  background: color,
-})
-
-const beforeAndAfter = (color, size, type, variables): ICSSPseudoElementStyle => ({
+const beforeAndAfter = (color, size, variables: DividerVariables): ICSSInJSStyle => ({
   content: '""',
   flex: 1,
-  ...dividerBorderStyle(size, variables.dividerColor),
-  ...(color && {
-    ...dividerBorderStyle(size, _.get(variables.colors, color)),
-  }),
+  height: `${size + 1}px`,
+  background: color ? _.get(variables.colors, color) : variables.dividerColor,
 })
 
-const dividerStyles: ComponentSlotStylesInput<DividerPropsWithDefaults, any> = {
+const dividerStyles: ComponentSlotStylesInput<DividerPropsWithDefaults, DividerVariables> = {
   root: ({ props, variables }): ICSSInJSStyle => {
-    const { children, color, fitted, size, type, important, content } = props
+    const { children, color, fitted, size, important, content } = props
     return {
-      color: variables.textColor,
+      color: color ? _.get(variables.colors, color) : variables.textColor,
```

Comment:
```suggestion
      color: _.get(variables.colors, color, variables.textColor),
```

This should work, too. However, I am not happy with this because `dividerColor` and `textColor` looks completely strange there 🤔 

---

It seems that we need to have a `default` color (may be `base`???) that will include this:
```diff
-type ColorNames = keyof (EmphasisColors & NaturalColors)
+type ColorNames = keyof (EmphasisColors & NaturalColors) & 'default'
```

And then we will able to do something like this.
```js
// color is `default` when is not defined

{
  color: _.get(variables.colors, [color, 'text'], 'inherit'), // inherit is a fallback value
  background: _.get(variables.colors, [color, 'background'], 'inherit'), // this thing can be our util in fact
}
```

Please let me know, what you think about this.

Reply:
@layershifter I made the change for

`color: _.get(variables.colors, color, variables.textColor)`

but for the default color I'm not even sure I get it, looks a bit like a premature optimization. Are you satisfied with current state?

Reply:
It seems that I wasn't very clear there. Take a look on `DividerVariables`, we have `colors`, `dividerColor` and `textColor` 🤔 

We can do this in a separate PR, but my proposal is to introduce a scheme there:
```diff
-type ColorNames = keyof (EmphasisColorsStrict & NaturalColorsStrict)
+type ColorNames = keyof (EmphasisColorsStrict & NaturalColorsStrict) & 'base'

//

interface DividerColorScheme {
  background: string
  text: string
}

export interface DividerVariables {
  colors: ColorValues<DividerColorScheme>
-  dividerColor: string
-  textColor: string
}
```

This will allow to drop `dividerColor` and `textColor`, they looks extremely confusing now.

Reply:
I'll make a proposal to introduce this to a separate PR, maybe we find a pattern for creating a unified scheme

Reply:
Yep, we discussed it yesterday. Marked this item as resolved.